protocols/imap/src/main/java/org/apache/james/imap/processor/ListProcessor.java

```diff
@@ -172,15 +175,19 @@ public MailboxId getId() {
                     basePath = PathConverter.forSession(session).buildFullPath(finalReferencename);
                 }
 
-                results = getMailboxManager().search(
+                PrefixedRegex prefixedRegex = new PrefixedRegex(
+                    basePath.getName(),
+                    CharsetUtil.decodeModifiedUTF7(mailboxName),
+                    mailboxSession.getPathDelimiter());
+                results = getMailboxManager()
```

Comment:
are we loading every single user mailboxes the user has access to then run a java filter on the collection ?

Reply:
Yes, but it's loaded anyway. And it is not that bad.

BTW we can speak of why this is needed ;-) I would send you back to PR description.

Reply:
I understand that Cassandra is already doing it, but what about the other backends? JPA for example?

Reply:
Yes, JPA will be less optimized.

That being said we don't know how to handle push back reference to make this both **correct** and **optimized** (push back reference if I remember @mbaechler  last daily speach). In that context, we prefered doing the **correct** choice, even at a performance penalty.

Reply:
>  In that context, we prefered doing the correct choice, even at a performance penalty.

"We" at least doesn't include "me". I agreed having same performance property for cassandra. I didn't agree to make JPA worse.